lcls-twincat-motion/Library/POUs/Motion/FB_MotionRequest.TcPOU

```diff
@@ -0,0 +1,154 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1" ProductVersion="3.1.4022.18">
+  <POU Name="FB_MotionRequest" Id="{f3ce72e2-b968-4260-9a64-8e0267e89e51}" SpecialFunc="None">
+    <Declaration><![CDATA[FUNCTION_BLOCK FB_MotionRequest
+(*
+	Request a move from an axis controlled via EPICS using FB_MotionStage
+	This exists to manage situations where different bits of code may need to move the same motor.
+	With just the DUT_MotionStage/FB_MotionStage setup it is possible for two function blocks to
+	fight with and interfere with each other and with the EPICS commands.
+*)
+VAR_IN_OUT
+	// Motor to move
+    stMotionStage: DUT_MotionStage;
+END_VAR
+VAR_INPUT
+	// Start move on rising edge, stop move on falling edge
+    bExecute: BOOL;
+	// Reset errors on rising edge
+    bReset: BOOL;
+	// Define behavior for when the motor is already moving
+    enumMotionRequest: ENUM_MotionRequest := ENUM_MotionRequest.WAIT;
+	// Goal position
+    fPos: LREAL;
+	// Move velocity
+    fVel: LREAL;
+	// Optional acceleration
+    fAcc: LREAL;
+	// Optional deceleration
+    fDec: LREAL;
+END_VAR
+VAR_OUTPUT
+	// True if in error state
+    bError: BOOL;
+	// Error code, usually
+    sErrorMessage: STRING;
+	// If TRUE, we are moving the motor
+    bBusy: BOOL;
+	// If TRUE, we are not moving the motor and our most recent move was successful
+    bDone: BOOL;
+END_VAR
+VAR
+    rtExec: R_TRIG;
+    ftExec: F_TRIG;
+    rtReset: R_TRIG;
+    ftBusy: F_TRIG;
+    nState: UINT := 0;
+    nQueuedState: UINT;
+    bMyMove: BOOL;
+END_VAR
+// Define local constants for our state machine states
+VAR CONSTANT
+    INIT: UINT := 0;
+    EXEC: UINT := 1;
+    START: UINT := 2;
+    MOVING: UINT := 3;
+    DONE: UINT := 4;
+    STOP: UINT := 5;
+    ERROR: UINT := 6;
+END_VAR
+]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[rtExec(CLK:=bExecute);
+ftExec(CLK:=bExecute);
+rtReset(CLK:=bReset);
+
+// Go back to INIT state on reset
+IF rtReset.Q THEN
+    nState := INIT;
+    nQueuedState := INIT;
+END_IF
+IF rtExec.Q OR ftExec.Q THEN
+	bDone := FALSE;
+END_IF
+
+CASE nState OF
+	// Start by setting everything to a known value
+    INIT:
+        nState := EXEC;
+        bError := FALSE;
+        sErrorMessage := '';
+        bBusy := FALSE;
+        bDone := FALSE;
+	// Normal "waiting for move" state
+    EXEC:
+        bMyMove := FALSE;
+		// Looking for a rising edge on bExecute
```

Comment:
I have a github issue for normalizing the whole repo at #32. I decided against doing it now because it would have changed lines in files I didn't touch and made the whole thing a pain to review.